operator/pkg/controllers/master/authenticatorconfig.go

```diff
@@ -35,146 +28,10 @@ func (c *Controller) reconcileAuthenticatorConfig(ctx context.Context, controlPl
-							}, {
-								Name:      "kubeconfig",
-								MountPath: "/var/aws-iam-authenticator/kubeconfig/",
-							}},
-						}},
-						Containers: []v1.Container{{
-							Name:  "aws-iam-authenticator",
-							Image: imageprovider.AWSIamAuthenticator(),
-							Args: []string{
-								"server",
-								"--master=https://localhost/",
-								"--config=/etc/aws-iam-authenticator/config.yaml",
-								"--state-dir=/var/aws-iam-authenticator/state/",
-								"--generate-kubeconfig=/var/aws-iam-authenticator/kubeconfig/kubeconfig.yaml",
-							},
-							SecurityContext: &v1.SecurityContext{AllowPrivilegeEscalation: ptr.Bool(true)},
-							VolumeMounts: []v1.VolumeMount{{
-								Name:      "config",
-								MountPath: "/etc/aws-iam-authenticator/",
-							}, {
-								Name:      "state",
-								MountPath: "/var/aws-iam-authenticator/state/",
-							}, {
-								Name:      "kubeconfig",
-								MountPath: "/var/aws-iam-authenticator/kubeconfig/",
-							}},
-						}},
-						Volumes: []v1.Volume{{
-							Name: "config",
-							VolumeSource: v1.VolumeSource{
-								ConfigMap: &v1.ConfigMapVolumeSource{
-									LocalObjectReference: v1.LocalObjectReference{Name: AuthenticatorConfigMapName(controlPlane.ClusterName())},
-								},
-							},
-						}, {
-							Name: "kubeconfig",
-							VolumeSource: v1.VolumeSource{
-								HostPath: &v1.HostPathVolumeSource{
-									Path: "/var/aws-iam-authenticator/kubeconfig/",
-								},
-							},
-						}, {
-							Name: "state",
-							VolumeSource: v1.VolumeSource{
-								HostPath: &v1.HostPathVolumeSource{
-									Path: "/var/aws-iam-authenticator/state/",
-								},
-							},
-						}},
-					},
-				},
-			},
-		},
-	)
-}
-
-func authenticatorLabels() map[string]string {
-	return map[string]string{
-		"k8s-app": "aws-iam-authenticator",
-	}
-}
-
-var (
-	authenticatorConfig = `
-apiVersion: v1
-kind: ConfigMap
-metadata:
-  name: {{ .Name }}
-  namespace: {{ .Namespace }}
-data:
-  config.yaml: |
-    clusterID: {{ .ClusterName }}.{{ .Group }}
-    server:
-      mapRoles:
-        - groups:
-          - system:bootstrappers
-          - system:nodes
-          rolearn: arn:aws:iam::{{ .AWSAccountID }}:role/{{ .KitNodeRole }}
-          username: system:node:{{ .PrivateDNS}}
-        - groups:
-          - system:bootstrappers
-          - system:nodes
-          rolearn: arn:aws:iam::{{ .AWSAccountID }}:role/KitletNodeRole
-          username: system:node:{{ .SessionName }}
-      # List of Account IDs to whitelist for authentication
-      mapAccounts:
-        - {{ .AWSAccountID }}
-`
-)
-func AuthenticatorDaemonSetName(clusterName string) string {
-	return fmt.Sprintf("%s-authenticator", clusterName)
-}
-
-func AuthenticatorConfigMapName(clusterName string) string {
-	return fmt.Sprintf("%s-auth-config", clusterName)
+	return c.awsIamAuthenticator.EnsureDaemonSet(ctx, controlPlane, APIServerLabels(controlPlane.ClusterName()))
```

Comment:
Seems odd to wrap this function if these are straight passthroughs. Why can't the caller be aware and dedup this?